### PR TITLE
spread tests: remove references of core16

### PR DESCRIPTION
**Pull request by cjp256**
Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: tests/spread/plugins/v1/dotnet/snaps/dotnet-hello/snap/snapcraft.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ description: |
   This is a sample dotnet 2.0.0 application
 
 grade: devel
-base: core16
+base: core18
```
Comment (sergiusens) [inline tests/spread/plugins/v1/dotnet/snaps/dotnet-hello/snap/snapcraft.yaml:8]: dotnet snap does not support core18

File: tests/spread/general/license-valid/task.yaml [hunk 2]
```diff
@@ -7,20 +7,20 @@ prepare: |
...
-      echo "base: core18" >> snap/snapcraft.yaml
-  elif [[ "$SPREAD_SYSTEM" =~ ubuntu-16.04 ]]; then
-      echo "base: core16" >> snap/snapcraft.yaml
-  else
-      FAIL "Test not supported for $SPREAD_SYSTEM"
-  fi 
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "../snaps/$SNAP/snap/snapcraft.yaml"
```
Comment (sergiusens) [inline tests/spread/general/license-valid/task.yaml:13]: the setup runs "snapcraft init" so the leading ../ is wrong

File: tests/spread/general/license-valid/task.yaml [hunk 3]
```diff
@@ -7,20 +7,20 @@ prepare: |
...
+
   echo "license: MIT" >> snap/snapcraft.yaml
 
 restore: |
   rm -rf test-snap
 
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
```
Comment (sergiusens) [inline tests/spread/general/license-valid/task.yaml:22]: the entire test-snap directory is wiped, this shouldn't be needed

Comment (cjp256) [reply tests/spread/general/license-valid/task.yaml:22]: Well that was clearly a blind copy and paste fail.

Fixed!

File: tests/spread/general/license-valid/task.yaml (hunk 2 above)
Comment (cjp256) [reply tests/spread/general/license-valid/task.yaml:13]: Fixed!

File: tests/spread/plugins/v1/dotnet/snaps/dotnet-hello/snap/snapcraft.yaml (hunk 1 above)
Comment (cjp256) [reply tests/spread/plugins/v1/dotnet/snaps/dotnet-hello/snap/snapcraft.yaml:8]: It built & ran for me just fine?

Comment (cjp256) [reply tests/spread/plugins/v1/dotnet/snaps/dotnet-hello/snap/snapcraft.yaml:8]: Confirmed again, seems fine..

Comment (cjp256) [reply tests/spread/plugins/v1/dotnet/snaps/dotnet-hello/snap/snapcraft.yaml:8]: There's also a stage-package handling for core18 in the plugin..

**Review by sergiusens — APPROVED** [review]

**Pull request merged**